perfkitbenchmarker/benchmark_sets.py

```diff
@@ -125,9 +130,18 @@ def GetBenchmarksFromFlags():
   If multiple sets or mixes of sets and benchmarks are specified, this will
   return the union of all sets and individual benchmarks.
   """
+  # Create a dictionary of valid benchmark names and modules
+  # TODO(voellm): I really think the benchmarks package init should
+  # build the dictionary
+  valid_benchmarks = {}
+  for benchmark_module in benchmarks.BENCHMARKS:
+    valid_benchmarks[benchmark_module.GetInfo()['name']] = benchmark_module
+
   benchmark_names = set()
   for benchmark in FLAGS.benchmarks:
-    if benchmark in BENCHMARK_SETS:
+    if benchmark == ALL_SET:
+      benchmark_names.update(valid_benchmarks)
+    elif benchmark in BENCHMARK_SETS:
```

Comment:
This is a bit tricky. It would have been nicer to have the ALL_SET entry in BENCHMARK_SETS contain the full BENCHMARK_LIST, but those values aren't available yet at the time BENCHMARK_SETS is assigned, it's only calculated a few lines above in the code block I moved. Could patch the value in at runtime, but it seems dubious to write to a module constant.

The loop iterates over the individual benchmark names provided by the user in the --benchmarks flag value list, and if it encounters the ALL_SET name it replaces that with the list of all known benchmarks. It's not equivalent to the recursive expansion below since the ALL_SET doesn't recursively expand to other sets.

Does that seem ok, or would you prefer me to change it to populate BENCHMARK_SETS[ALL_SET][BENCHMARK_LIST] via patching it in initialization?